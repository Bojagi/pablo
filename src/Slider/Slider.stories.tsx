import React, { useState } from 'react';

import { Slider } from './Slider';
import { Flex } from '../Box';
import { Paragraph } from '../Typography';

export default {
  title: 'Slider',
};

const BaseStory = ({ from, to }) => {
  const [value, setValue] = useState(50);
  return (
    <Flex center width="50vw" gap={2}>
      <Slider shrink grow from={from} to={to} value={value} onChange={setValue} />
      <Flex width={50} center shrink={0} grow={0}>
        <Paragraph inline>{Math.floor(value)}</Paragraph>
      </Flex>
    </Flex>
  );
};

export const SimpleSlider = BaseStory.bind(null);
SimpleSlider.args = {
  from: 0,
  to: 100,
};

export const OffsetRangeSlider = BaseStory.bind(null);
OffsetRangeSlider.args = {
  from: 23,
  to: 184,
};

export const LowerOutOfBounds = BaseStory.bind(null);
LowerOutOfBounds.args = {
  from: 70,
  to: 100,
};

export const UpperOutOfBounds = BaseStory.bind(null);
UpperOutOfBounds.args = {
  from: 0,
  to: 40,
};
