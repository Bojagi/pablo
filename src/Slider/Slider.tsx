import React, { MouseEventHandler, useRef, useState } from 'react';
import { LayoutBox, LayoutBoxProps } from '../Box';
import styled from '@emotion/styled';
import { useEventListener } from '../utils/useEventListener';
import { getComponentStyle, shadowTransformer } from '../styleHelpers';
import { componentPrimitive, getPrimitiveStyle } from '../styleHelpers/componentPrimitive';

interface SliderProps extends LayoutBoxProps {
  from: number;
  to: number;
  value: number;
  onChange: (value: number) => void;
}

const thickness = getPrimitiveStyle('thickness');
const borderRadius = getPrimitiveStyle('borderRadius');
const backgroundColor = getPrimitiveStyle('backgroundColor');

const Rail = componentPrimitive(['slider', 'rail'])`
  width: 100%;
  height: ${thickness};
  border-radius: ${borderRadius};
  background-color: ${backgroundColor};
`;

const Track = componentPrimitive(['slider', 'track'])`
  position: absolute;
  height: ${thickness};
  top: 50%;
  transform: translateY(-50%);
  border-radius: ${borderRadius};
  background-color: ${backgroundColor};
`;

const Thumb = componentPrimitive(['slider', 'thumb'])`
  width: ${getPrimitiveStyle('width')};
  height: ${getPrimitiveStyle('height')};
  border-radius: ${borderRadius};
  background-color: ${backgroundColor};
  box-shadow: ${getPrimitiveStyle('shadow', shadowTransformer)};
  position: absolute;
  top: 50%;
  transform: translateY(-50%) translateX(-50%);
  cursor: grab;
`;

const OuterBox = styled(LayoutBox)`
  width: 100%;
  height: ${getComponentStyle(['slider', 'thumb', 'height'])};
  position: relative;
  display: flex;
  align-items: center;
`;

const Slider = ({ from, to, value, onChange, ...props }: SliderProps) => {
  const offsetRef = useRef<number>(0);
  const railRef = useRef<HTMLDivElement>(null);
  const [dragging, setDragging] = useState(false);
  const ratio = Math.min(Math.max((value - from) / (to - from), 0), 1);
  const percentage = ratio * 100;

  useEventListener('mouseup', () => setDragging(false));
  useEventListener(
    'mousemove',
    (e: MouseEvent) => {
      if (dragging && railRef.current) {
        const { left, width } = railRef.current.getBoundingClientRect();
        const x = e.clientX - left + offsetRef.current;
        const ratio = x / width;
        const calculatedValue = from + ratio * (to - from);
        const newValue = Math.max(Math.min(calculatedValue, to), from);
        onChange(newValue);
      }
    },
    [dragging, onChange, from, to]
  );

  const handleThumbMouseDown: MouseEventHandler<HTMLDivElement> = (e) => {
    e.preventDefault();
    const target = e.target as HTMLDivElement;
    const boundingRect = target.getBoundingClientRect();
    offsetRef.current = boundingRect.left + boundingRect.width * 0.5 - e.clientX;

    setDragging(true);
  };

  return (
    <OuterBox {...props} role="slider">
      <Rail data-testid="slider-rail" ref={railRef} />
      <Track data-testid="slider-track" style={{ width: `${percentage}%` }} />
      <Thumb
        data-testid="slider-thumb"
        onMouseDown={handleThumbMouseDown}
        style={{ left: `${percentage}%` }}
      />
    </OuterBox>
  );
};

export { Slider };
