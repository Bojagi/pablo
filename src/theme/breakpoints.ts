export type Breakpoint = 'sm' | 'md' | 'lg' | 'xl';

export const breakpointNames: Breakpoint[] = ['sm', 'md', 'lg', 'xl'];

export type BreakpointsArray = [number, number, number, number];

export type Breakpoints = BreakpointsArray &
  Record<Breakpoint, number> & {
    breakpointNames: Breakpoint[];
  };

export function createBreakpoints(
  array: BreakpointsArray,
  usedBreakpointNames: Breakpoint[] = breakpointNames
): Breakpoints {
  const newBreakpoints = ([...array] as unknown) as Breakpoints;
  usedBreakpointNames.forEach((bp, index) => {
    newBreakpoints[bp] = newBreakpoints[index];
  });
  newBreakpoints.breakpointNames = usedBreakpointNames;

  return newBreakpoints as Breakpoints;
}

export const defaultBreakpoints: BreakpointsArray = [600, 1000, 1200, 1920];

export const breakpoints = createBreakpoints(defaultBreakpoints, breakpointNames);
