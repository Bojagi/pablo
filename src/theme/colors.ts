export type ShadedColor = {
  main: string;
  dark: string;
  darkest: string;
  light: string;
  lightest: string;
  contrastText: string;
  contrastTextLight: string;
};

export type CommonColors = {
  white: string;
  black: string;
  whiteContrastText: string;
  blackContrastText: string;
};

export type TextColors = {
  main: string;
  info: string;
};

export type BorderColors = {
  main: string;
  lightest: string;
};

export interface GrayColors {
  '50': string;
  '100': string;
  '200': string;
  '300': string;
  '400': string;
  '500': string;
  '600': string;
  '700': string;
  '800': string;
  '900': string;
}

export type AllColors = ShadedColor & TextColors & BorderColors & CommonColors;

export interface Colors {
  common: CommonColors;
  gray: GrayColors;
  borders: BorderColors;
  brand: ShadedColor;
  positive: ShadedColor;
  negative: ShadedColor;
  neutral: ShadedColor;
  text: TextColors;
}

export const WHITE = '#ffffff';
export const BLACK = '#000000';

export const colors: Colors = {
  common: {
    white: WHITE,
    black: BLACK,
    whiteContrastText: BLACK,
    blackContrastText: WHITE,
  },
  gray: {
    '50': getGrayColor(0.95),
    '100': getGrayColor(0.9),
    '200': getGrayColor(0.8),
    '300': getGrayColor(0.7),
    '400': getGrayColor(0.6),
    '500': getGrayColor(0.5),
    '600': getGrayColor(0.4),
    '700': getGrayColor(0.3),
    '800': getGrayColor(0.2),
    '900': getGrayColor(0.1),
  },
  borders: {
    main: `${BLACK}${getHexByte(0.25)}`,
    lightest: `${BLACK}${getHexByte(0.1)}`,
  },
  text: {
    main: BLACK,
    info: `${BLACK}${getHexByte(0.5)}`,
  },
  brand: {
    lightest: '#EBE7FF',
    light: '#D6CEFF',
    main: '#6A50F2',
    dark: '#4129BD',
    darkest: '#281683',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  positive: {
    lightest: '#DFF6E6',
    light: '#BBECCB',
    main: '#38C765',
    dark: '#2FB85A',
    darkest: '#108B36',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  negative: {
    lightest: '#FFE7E5',
    light: '#FFCFCC',
    main: '#F21D0D',
    dark: '#CA271B',
    darkest: '#871A12',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
  neutral: {
    lightest: '#FCF5DB',
    light: '#FFEB9E',
    main: '#FFDD5B',
    dark: '#EDC52A',
    darkest: '#D6AC0B',
    contrastText: WHITE,
    contrastTextLight: BLACK,
  },
};

/**
 * Get hex color byte (e.g. to get transparency)
 * @param value Percentage value (between 0.0 and 1.0)
 */
function getHexByte(value: number) {
  return (Math.round((255 * value) / 16) * 16).toString(16);
}

function getGrayColor(value: number) {
  return `#${getHexByte(value)}${getHexByte(value)}${getHexByte(value)}`;
}
