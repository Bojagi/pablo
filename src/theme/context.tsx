import * as React from 'react';
import { PabloTheme, ComponentStyles } from './types';

export const pabloThemeContext = React.createContext<PabloTheme>({} as any);
export const pabloComponentStylesContext = React.createContext<ComponentStyles>({} as any);
export const rootContext = React.createContext<Document | ShadowRoot | null>(null);

export const useComponentStyleContext = () => React.useContext(pabloComponentStylesContext);

export const usePabloTheme = () => React.useContext(pabloThemeContext);
