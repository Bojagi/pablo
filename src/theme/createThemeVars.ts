import isObject from 'isobject';

type DeepStringReplace<T, M extends [any, any]> = {
  [P in keyof T]: T[P] extends object ? DeepStringReplace<T[P], M> : string;
};

export function createThemeVars<T extends Record<string, any>>(base: string, themeObject: T) {
  return internalThemeVars(`pbl-theme-${base}`, themeObject);
}

function internalThemeVars<T extends Record<string, any>>(
  base: string,
  themeObject: T
): DeepStringReplace<T, [number, string]> {
  return Object.entries(themeObject).reduce((acc, [key, value]) => {
    const keyVar = `${base}-${key}`;
    return { ...acc, [key]: isObject(value) ? internalThemeVars(keyVar, value) : `${keyVar}` };
  }, {} as any);
}

export function createThemeVarArray<T extends readonly string[]>(
  base: string,
  values: T
): Record<T[number], string> {
  return values.reduce(
    (acc, value) => ({ ...acc, [value]: `pbl-theme-${base}-${value}` }),
    {} as any
  );
}
