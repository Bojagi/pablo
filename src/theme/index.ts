export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
