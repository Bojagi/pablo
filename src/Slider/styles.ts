import { css } from '@emotion/react';
import type * as CSS from 'csstype';
import { getComponentStyle } from '../styleHelpers';
import { themeVars } from '../theme/themeVars';
import { ComponentPrimitiveStyle, Style } from '../theme/types';
import { BaseStyles } from '../types';

export type SliderStyleProperties = 'root';

export interface SliderRailStyles extends ComponentPrimitiveStyle {
  borderRadius: Style;
  backgroundColor: Style;
  thickness: Style;
}

export interface SliderThumbStyles extends ComponentPrimitiveStyle {
  borderRadius: Style;
  backgroundColor: Style;
  width: Style;
  height: Style;
  shadow: string[];
  css: Style;
  cursor: CSS.Property.Cursor;
}

export interface SliderTrackStyles extends ComponentPrimitiveStyle {
  backgroundColor: Style;
  thickness: Style;
  borderRadius: Style;
}

export interface SliderStyles extends BaseStyles<SliderStyleProperties> {
  rail: SliderRailStyles;
  thumb: SliderThumbStyles;
  track: SliderTrackStyles;
}

export const sliderStyles: SliderStyles = {
  rail: {
    borderRadius: getComponentStyle(['slider', 'rail', 'thickness']),
    backgroundColor: themeVars.colors.gray[100],
    thickness: '8px',
  },
  thumb: {
    borderRadius: '50%',
    backgroundColor: themeVars.colors.common.white,
    height: '16px',
    width: getComponentStyle(['slider', 'thumb', 'height']),
    shadow: ['0px 1px 2px rgba(0, 0, 0, 0.2)', '0px 4px 10px rgba(0, 0, 0, 0.1)'],
    css: css`
      border: 1px solid ${themeVars.colors.gray[100]};
    `,
    cursor: 'grab',
  },
  track: {
    backgroundColor: themeVars.colors.brand.main,
    thickness: '10px',
    borderRadius: getComponentStyle(['slider', 'track', 'thickness']),
  },
};
