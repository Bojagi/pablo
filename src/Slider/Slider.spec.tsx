import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { Slider } from './Slider';
import * as stories from './Slider.stories';
import { PabloThemeProvider } from '../theme';

const defaultProps = {
  from: 0,
  to: 100,
  value: 50,
  onChange: vi.fn(),
};

beforeEach(() => {
  vi.clearAllMocks();
});

test('renders correctly', () => {
  render(<Slider {...defaultProps} />);
  expect(screen.getByRole('slider')).toBeInTheDocument();
});

test('calls onChange when dragging thumb', () => {
  const { container } = render(<Slider {...defaultProps} />);
  const thumb = container.querySelector('[data-testid="slider-thumb"]');
  const rail = container.querySelector('[data-testid="slider-rail"]');

  if (!thumb || !rail) throw new Error('Thumb or rail not found');

  // Mock getBoundingClientRect
  const railRect = { left: 0, width: 200 };
  vi.spyOn(rail, 'getBoundingClientRect').mockImplementation(() => railRect as DOMRect);
  // Set bounding rect that the click is exactly in the center
  jest
    .spyOn(thumb, 'getBoundingClientRect')
    .mockImplementation(() => ({ left: 90, width: 20 }) as DOMRect);

  // Simulate drag start
  fireEvent.mouseDown(thumb, { clientX: 100 });

  // Simulate drag movement
  fireEvent.mouseMove(window, { clientX: 150 });

  expect(defaultProps.onChange).toHaveBeenCalledWith(75);
});

test('handles custom range correctly', () => {
  const customProps = {
    ...defaultProps,
    from: -100,
    to: 100,
    value: 0,
  };

  const { container } = render(<Slider {...customProps} />);
  const thumb = container.querySelector('[data-testid="slider-thumb"]');
  const rail = container.querySelector('[data-testid="slider-rail"]');

  if (!thumb || !rail) throw new Error('Thumb or rail not found');

  const railRect = { left: 0, width: 200 };
  vi.spyOn(rail, 'getBoundingClientRect').mockImplementation(() => railRect as DOMRect);
  // Set bounding rect that the click is exactly in the center
  jest
    .spyOn(thumb, 'getBoundingClientRect')
    .mockImplementation(() => ({ left: 90, width: 20 }) as DOMRect);

  fireEvent.mouseDown(thumb, { clientX: 100 });
  fireEvent.mouseMove(window, { clientX: 150 });

  expect(customProps.onChange).toHaveBeenCalledWith(50);
});

test('constrains value within bounds', () => {
  const { container } = render(<Slider {...defaultProps} />);
  const thumb = container.querySelector('[data-testid="slider-thumb"]');
  const rail = container.querySelector('[data-testid="slider-rail"]');

  if (!thumb || !rail) throw new Error('Thumb or rail not found');

  // Mock getBoundingClientRect
  const railRect = { left: 0, width: 200 };
  vi.spyOn(rail, 'getBoundingClientRect').mockImplementation(() => railRect as DOMRect);

  // Simulate drag beyond maximum
  fireEvent.mouseDown(thumb, { clientX: 100 });
  fireEvent.mouseMove(document, { clientX: 300 });

  expect(defaultProps.onChange).toHaveBeenCalledWith(100);

  // Simulate drag below minimum
  fireEvent.mouseMove(document, { clientX: -100 });

  expect(defaultProps.onChange).toHaveBeenCalledWith(0);
});

test('stops dragging on mouseup', () => {
  const { container } = render(<Slider {...defaultProps} />);
  const thumb = container.querySelector('[data-testid="slider-thumb"]');

  if (!thumb) throw new Error('Thumb not found');

  fireEvent.mouseDown(thumb);
  fireEvent.mouseUp(document);

  // Simulate move after mouseup
  fireEvent.mouseMove(document, { clientX: 150 });

  expect(defaultProps.onChange).not.toHaveBeenCalled();
});

test('updates track width based on value', () => {
  render(<Slider {...defaultProps} value={25} />);
  const track = screen.getByTestId('slider-track');
  expect(track).toHaveStyle({ width: '25%' });
});

test('updates thumb position based on value', () => {
  render(<Slider {...defaultProps} value={75} />);
  const thumb = screen.getByTestId('slider-thumb');
  expect(thumb).toHaveStyle({ left: '75%' });
});

test('SimpleSlider story snapshot', () => {
  const { container } = render(createStoryComponent(stories.SimpleSlider));
  expect(container.firstChild).toMatchSnapshot();
});

test('OffsetRangeSlider story snapshot', () => {
  const { container } = render(createStoryComponent(stories.OffsetRangeSlider));
  expect(container.firstChild).toMatchSnapshot();
});

test('LowerOutOfBounds story snapshot', () => {
  const { container } = render(createStoryComponent(stories.LowerOutOfBounds));
  expect(container.firstChild).toMatchSnapshot();
});

test('UpperOutOfBounds story snapshot', () => {
  const { container } = render(createStoryComponent(stories.UpperOutOfBounds));
  expect(container.firstChild).toMatchSnapshot();
});

function createStoryComponent(Story) {
  return (
    <PabloThemeProvider>
      <Story {...Story.args} />
    </PabloThemeProvider>
  );
}
