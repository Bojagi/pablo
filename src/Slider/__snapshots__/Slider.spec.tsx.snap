// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LowerOutOfBounds story snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  gap: 16px 16px;
  width: 50vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: 100%;
  height: 16px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  width: 100%;
  height: 8px;
  border-radius: 8px;
  background-color: var(--pbl-theme-colors-gray-100);
}

.emotion-3 {
  position: absolute;
  height: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  border-radius: 10px;
  background-color: var(--pbl-theme-colors-brand-main);
}

.emotion-4 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: var(--pbl-theme-colors-common-white);
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.2),0px 4px 10px rgba(0, 0, 0, 0.1);
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  -moz-transform: translateY(-50%) translateX(-50%);
  -ms-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
  cursor: -webkit-grab;
  cursor: grab;
  border: 1px solid var(--pbl-theme-colors-gray-100);
}

.emotion-5 {
  box-sizing: border-box;
  width: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-style: normal;
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin: 0;
  font-size: var(--pbl-theme-typography-paragraph-fontSize);
  line-height: var(--pbl-theme-typography-paragraph-lineHeight);
  margin-bottom: 0;
}

<div
  class="emotion-0"
  width="50vw"
>
  <div
    class="emotion-1"
    role="slider"
  >
    <div
      class="emotion-2"
      data-testid="slider-rail"
    />
    <div
      class="emotion-3"
      data-testid="slider-track"
      style="width: 0%;"
    />
    <div
      class="emotion-4"
      data-testid="slider-thumb"
      style="left: 0%;"
    />
  </div>
  <div
    class="emotion-5"
    width="50"
  >
    <p
      class="emotion-6"
    >
      50
    </p>
  </div>
</div>
`;

exports[`OffsetRangeSlider story snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  gap: 16px 16px;
  width: 50vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: 100%;
  height: 16px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  width: 100%;
  height: 8px;
  border-radius: 8px;
  background-color: var(--pbl-theme-colors-gray-100);
}

.emotion-3 {
  position: absolute;
  height: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  border-radius: 10px;
  background-color: var(--pbl-theme-colors-brand-main);
}

.emotion-4 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: var(--pbl-theme-colors-common-white);
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.2),0px 4px 10px rgba(0, 0, 0, 0.1);
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  -moz-transform: translateY(-50%) translateX(-50%);
  -ms-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
  cursor: -webkit-grab;
  cursor: grab;
  border: 1px solid var(--pbl-theme-colors-gray-100);
}

.emotion-5 {
  box-sizing: border-box;
  width: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-style: normal;
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin: 0;
  font-size: var(--pbl-theme-typography-paragraph-fontSize);
  line-height: var(--pbl-theme-typography-paragraph-lineHeight);
  margin-bottom: 0;
}

<div
  class="emotion-0"
  width="50vw"
>
  <div
    class="emotion-1"
    role="slider"
  >
    <div
      class="emotion-2"
      data-testid="slider-rail"
    />
    <div
      class="emotion-3"
      data-testid="slider-track"
      style="width: 16.77018633540373%;"
    />
    <div
      class="emotion-4"
      data-testid="slider-thumb"
      style="left: 16.77018633540373%;"
    />
  </div>
  <div
    class="emotion-5"
    width="50"
  >
    <p
      class="emotion-6"
    >
      50
    </p>
  </div>
</div>
`;

exports[`SimpleSlider story snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  gap: 16px 16px;
  width: 50vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: 100%;
  height: 16px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  width: 100%;
  height: 8px;
  border-radius: 8px;
  background-color: var(--pbl-theme-colors-gray-100);
}

.emotion-3 {
  position: absolute;
  height: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  border-radius: 10px;
  background-color: var(--pbl-theme-colors-brand-main);
}

.emotion-4 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: var(--pbl-theme-colors-common-white);
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.2),0px 4px 10px rgba(0, 0, 0, 0.1);
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  -moz-transform: translateY(-50%) translateX(-50%);
  -ms-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
  cursor: -webkit-grab;
  cursor: grab;
  border: 1px solid var(--pbl-theme-colors-gray-100);
}

.emotion-5 {
  box-sizing: border-box;
  width: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-style: normal;
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin: 0;
  font-size: var(--pbl-theme-typography-paragraph-fontSize);
  line-height: var(--pbl-theme-typography-paragraph-lineHeight);
  margin-bottom: 0;
}

<div
  class="emotion-0"
  width="50vw"
>
  <div
    class="emotion-1"
    role="slider"
  >
    <div
      class="emotion-2"
      data-testid="slider-rail"
    />
    <div
      class="emotion-3"
      data-testid="slider-track"
      style="width: 50%;"
    />
    <div
      class="emotion-4"
      data-testid="slider-thumb"
      style="left: 50%;"
    />
  </div>
  <div
    class="emotion-5"
    width="50"
  >
    <p
      class="emotion-6"
    >
      50
    </p>
  </div>
</div>
`;

exports[`UpperOutOfBounds story snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  gap: 16px 16px;
  width: 50vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: 100%;
  height: 16px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  width: 100%;
  height: 8px;
  border-radius: 8px;
  background-color: var(--pbl-theme-colors-gray-100);
}

.emotion-3 {
  position: absolute;
  height: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -moz-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  border-radius: 10px;
  background-color: var(--pbl-theme-colors-brand-main);
}

.emotion-4 {
  width: 16px;
  height: 16px;
  border-radius: 50%;
  background-color: var(--pbl-theme-colors-common-white);
  box-shadow: 0px 1px 2px rgba(0, 0, 0, 0.2),0px 4px 10px rgba(0, 0, 0, 0.1);
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(-50%);
  -moz-transform: translateY(-50%) translateX(-50%);
  -ms-transform: translateY(-50%) translateX(-50%);
  transform: translateY(-50%) translateX(-50%);
  cursor: -webkit-grab;
  cursor: grab;
  border: 1px solid var(--pbl-theme-colors-gray-100);
}

.emotion-5 {
  box-sizing: border-box;
  width: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-style: normal;
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  margin: 0;
  font-size: var(--pbl-theme-typography-paragraph-fontSize);
  line-height: var(--pbl-theme-typography-paragraph-lineHeight);
  margin-bottom: 0;
}

<div
  class="emotion-0"
  width="50vw"
>
  <div
    class="emotion-1"
    role="slider"
  >
    <div
      class="emotion-2"
      data-testid="slider-rail"
    />
    <div
      class="emotion-3"
      data-testid="slider-track"
      style="width: 100%;"
    />
    <div
      class="emotion-4"
      data-testid="slider-thumb"
      style="left: 100%;"
    />
  </div>
  <div
    class="emotion-5"
    width="50"
  >
    <p
      class="emotion-6"
    >
      50
    </p>
  </div>
</div>
`;
